src/Pim/index.tsx

```diff
@@ -251,7 +253,7 @@ class Pim extends React.PureComponent {
     if (last) {
       prevUid = last.uid;
     }
-    store.dispatch<any>(fetchEntries(this.props.etesync, journal.uid, prevUid))
+    return store.dispatch<any>(fetchEntries(this.props.etesync, journal.uid, prevUid))
```

Comment:
According with my comment above, this function should not early return. Just let it return like before (by getting to the end of the function). Same with the line below.

Reply:
But then how can we await the finishing of the asynchronous action?

Reply:
Sorry, I thought the the function was async. Either make the function async and change the `return Promise.reject()` to `throw new Error('journalUid is undefined')` and await on these two async calls (and everything else will just work). Or you can manually create a promise. Though I think making the function async is much easier.

Reply:
To be honest though, the returns you did are fine too if changing the return type to `Promise<void>` won't complain about the types.

Reply:
Yeah TypeScript doesn't seem to have a problem with the return type. `store.dispatch` is weird, it returns whatever you pass in, which in this case is of the type `Action<Promise<EteSync.Entry[]>>`. Which I guess resolves to a Promise? I'm not sure, it's a bit confusing, but it works haha.

Honestly I am ok with either way (returning promises or marking it async), it seems more like a style choice at this point. Which would you prefer?

Reply:
It's typing issue which I haven't managed to solve. The return type is indeed as you said, but the actual return type is `Promise<Action<EteSync.Entry[]>>`. That's why there's an `<any>` there to workaround this issue. :|
If this works then no need to change it, it's all the same.